Marlin/src/HAL/STM32/HAL_SPI.cpp

```diff
@@ -78,7 +78,9 @@ static SPISettings spiConfig;
       case SPI_SPEED_6:      delaySPIFunc = &delaySPI_2000; break;  // desired:   250,000  actual: ~210K
       default:               delaySPIFunc = &delaySPI_4000; break;  // desired:   125,000  actual: ~123K
     }
-    SPI.begin();
+    #ifndef STM32G0B1xx
```

Comment:
In fact, this is not for `STM32G0B1xx`, but for all STM32 software SPI. When I first adapted `STM32G0B1xx` to Marlin, in order to reduce the impact of `Arduino_Core_STM32` library, I used the software SPI and found the problem here, detail in #23064, I think it should be deleted directly here, and I test it in Octopus and SKR-mini-E3-V3.0, but I'm not sure whether it will affect other products, so I didn't submit PR but submit an issue. 

SKR-mini-E3-V3.0 use the hardware SPI now. So here for SKR-mini-E3-V3.0 has no effect.